src/System.Net.Primitives/tests/FunctionalTests/Resources/System.Net.Primitives.Functional.Tests.rd.xml

```diff
@@ -0,0 +1,7 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<Directives xmlns="http://schemas.microsoft.com/netfx/2013/01/metadata">
+  <Library Name="System.Net.Primitives.Functional.Tests">
+    <Type Name="System.Net.CookieVariant" BinaryFormatter="All" />
+    <Type Name="System.Net.CookieCollection" BinaryFormatter="All" />
+  </Library>
+</Directives>
```

Comment:
Short version: Our tests are harder to make work without rd.xml than customer code will be.

Long version: As with any reflection, the .NET Native compiler needs to know what to keep for BinaryFormatter. If it keeps all serializable types in all apps, that will add considerable size, especially for apps that don't use BinaryFormatter at all. Of course, we generally consider it a failure when customers have to add rd.xml, so we've added BinaryFormatter-specific analysis to the .NET Native toolchain that understands common patterns like
```
BinaryFormatter bf = new BinaryFormatter();
bf.Serialize(someObject);
```
That includes cases like calling through wrapper methods etc. However, these and other test cases look like:
```
        public static IEnumerable<object[]> SerializeDeserialize_Roundtrip_MemberData()
        {
            yield return new object[] { new Cookie("somekey", "somevalue") };
            yield return new object[] { new CookieCollection() { new Cookie("somekey", "somevalue") } };
        }
        [Theory]
        [MemberData(nameof(SerializeDeserialize_Roundtrip_MemberData))]
        public static void SerializeDeserialize_Roundtrip_EqualObjects(object obj)
        {
            Assert.Equal(obj, BinaryFormatterHelpers.Clone(obj));
        }
```

So there's not really a straight line between the inputs to Clone and the objects returned by  SerializeDeserialize_Roundtrip_MemberData. Since we mostly don't expect customers to build code that indirect in most cases, we're opting for rd.xml in the test.